File: backend/collect/entity/ServiceLink.py
from sqlalchemy import Column, Integer, Boolean, TIMESTAMP, func

from entity.Base import Base


class ServiceLink(Base):
    __tablename__ = 'service_link'
    service_link_id = Column(Integer, primary_key=True, autoincrement=True)
    service_id = Column(Integer, nullable=True)
    linked_service_id = Column(Integer, nullable=False)
    enabled = Column(Boolean, nullable=False)
    created_at = Column(TIMESTAMP, nullable=False, default=func.now())
    updated_at = Column(TIMESTAMP, default=func.now(), onupdate=func.now())

File: backend/core/src/main/java/io/omegi/core/project/domain/ServiceLink.java
package io.omegi.core.project.domain;

import static jakarta.persistence.FetchType.*;
import static jakarta.persistence.GenerationType.*;
import static lombok.AccessLevel.*;

import java.time.LocalDateTime;

import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import jakarta.persistence.Entity;
import jakarta.persistence.EntityListeners;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "service_link")
@EntityListeners(AuditingEntityListener.class)
@Getter
@NoArgsConstructor(access = PROTECTED)
public class ServiceLink {

	@Id
	@GeneratedValue(strategy = IDENTITY)
	private Integer serviceLinkId;

	@ManyToOne(fetch = LAZY)
	@JoinColumn(name = "service_id")
	private Service service;

	@ManyToOne(fetch = LAZY)
	@JoinColumn(name = "linked_service_id")
	private Service linkedService;

	private boolean enabled;

	@CreatedDate
	private LocalDateTime createdAt;

	@LastModifiedDate
	private LocalDateTime updatedAt;
}
